### ISSUE TITLE
Mono 3.0.3rc1: Core API hash mismatch!

### ISSUE THREAD

**Issue by exts** [godotengine/godot#18593]
_(tried searching all issues tagged 3.0.3 nothing popped up)_

**Godot version:**
3.0.3rc1

**OS/device including version:**
Windows 10

**Issue description:**
Starting up godot 3.0.3rc1 in the console this error appears

```
OpenGL ES 3.0 Renderer: GeForce GTX 970/PCIe/SSE2
Mono: Initializing module...
ERROR: GDMono::_initialize_and_check_api_hashes: Mono: Core API hash mismatch!
   At: modules\mono\mono_gd\gd_mono.cpp:318
ERROR: GDMono::_initialize_and_check_api_hashes: Mono: Editor API hash mismatch!
   At: modules\mono\mono_gd\gd_mono.cpp:327
Mono: Runtime initialized
Mono: INITIALIZED
```

**Steps to reproduce:**
- I uninstalled mono 5.4.1
- I installed mono 5.10.1 Stable (5.10.1.47) x64
- Restarted PC
- Restarted Godot 3.0.3rc1

**Comment by hpvb** [comment]
This turned out to be an error in the build scripts. This should be corrected soon, sorry for the inconvenience!

**Comment by exts** [comment]
Yeah this completely breaks scripts it seems.

```
ERROR: GDMono::initialize: The loaded Core API assembly is out of sync
   At: modules\mono\mono_gd\gd_mono.cpp:253
ERROR: GDMono::initialize: The loaded Editor API assembly is out of sync
   At: modules\mono\mono_gd\gd_mono.cpp:261
Mono: Proceeding to unload scripts domain because of invalid API assemblies
```

Is an error that spits out when trying to attach a simple script to a scene.
[TestingGlobals.zip](https://github.com/godotengine/godot/files/1971919/TestingGlobals.zip)

**Comment by exts** [comment]
Just to let you guys know @akien-mga @neikeq this is a show stopper issue. You cannot build or test RC1 mono for 3.0.3 as of right now. Any ETA on a rebuild of RC1 mono?

**Comment by hpvb** [comment]
This is fixed in 3.0.3-rc2. Closing this issue.

**Comment by Marshal313** [comment]
I built from master and I use Mono 5.12.0.226, But it's the same for me. It gives me the following errors :

```
0:00:00:0719 - Mono Cache: Member GDMonoUtils::mono_cache.class_RemoteSyncAttribute is null
----------
Type:Error
Description: Mono Cache: Member GDMonoUtils::mono_cache.class_RemoteSyncAttribute is null
Time: 0:00:00:0719
C Error: Method/Function Failed.
C Source: modules/mono/mono_gd/gd_mono_utils.cpp:217
C Function: update_godot_api_cache
```

```
0:00:00:0721 - The loaded Core API assembly is out of sync
----------
Type:Error
Description: 
Time: 0:00:00:0721
C Error: The loaded Core API assembly is out of sync
C Source: modules/mono/mono_gd/gd_mono.cpp:247
C Function: initialize
```
